### PR TITLE
Add support for oracle invisible columns

### PR DESCRIPTION
**Pull request by imranzaheer612**
- Add support for Invisible Columns
- Add support for invisible columns in psql extended describle (\d+), in oracle mode only
- Changed Regression to for (\d+)
- Add regression for invisible columns

I used this patch https://www.postgresql.org/message-id/flat/be81d947-f4fe-c062-f107-0f05f8f87ca8%40migops.com and changed it accordingly to support ivorysql in oracle mode.

The invisible columns support in describe command (\d+) is causing most of the test cases to change. I feel this may add difficulty during the upstream merges. If we don't need this, we can remove the invisible attribute from the describe command.

I didn't add the test cases with MODIFY col commands. Looks like its not working. I opened the issue here https://github.com/IvorySQL/IvorySQL/issues/678. I will try fixing that too.

Please let me know if we need to change some thing in this PR.

This PR is related to https://github.com/IvorySQL/IvorySQL/issues/608

## Summary by CodeRabbit

- **New Features**
	- Introduced support for invisible columns in PostgreSQL tables.
	- Added operations for creating, altering, and querying tables with invisible columns.
	- Enhanced inheritance, partitioning, and temporary table functionalities to handle invisible columns.
	- Included new SQL commands and options for managing invisible columns.
	- Expanded capabilities for defining constraints and views with invisible columns.
	- Improved command completion options for altering table structures in the command line interface.
	- Enhanced the `pg_dump` tool to recognize and handle invisible columns in schema dumps.
	- Added comprehensive test cases to showcase various scenarios involving invisible columns.
- **Bug Fixes**
	- Improved handling of column visibility during star expansion in SQL queries.
	- Corrected logic for attribute processing to accurately filter invisible columns.

### CODE REVIEW

File: src/oracle_test/regress/expected/compression.out [hunk 1]
```diff
@@ -137,48 +137,48 @@ DROP TABLE cmdata2;
...
- Column |       Type        | Collation | Nullable | Default | Storage | Compression | Stats target | Description 
---------+-------------------+-----------+----------+---------+---------+-------------+--------------+-------------
- f1     | character varying |           |          |         | plain   | pglz        |              | 
+                                                  Table "public.cmdata2"
+ Column |      Type      | Collation | Nullable | Default | Invisible | Storage | Compression | Stats target | Description 
+--------+----------------+-----------+----------+---------+-----------+---------+-------------+--------------+-------------
+ f1     | varchar2(4000) |           |          |         |           | plain   | pglz        |              | 
 
 INSERT INTO cmdata2 VALUES (repeat('123456789', 800));
+ERROR:  value too long for type varchar2(4000 byte)
```
Comment (imranzaheer612) [inline src/oracle_test/regress/expected/compression.out:177]: Is this behavior is expected in oracle mode?

**Comment by imranzaheer612** [comment]
@gaoxueyu can you assign someone to review the PR?

**Comment by gaoxueyu** [comment]
> @gaoxueyu can you assign someone to review the PR?
Hi Imran，thanks for your great job. Jiaoshuntian is working on this.

**Comment by imranzaheer612** [comment]
> > * Add support for Invisible Columns
> > * Add support for invisible columns in psql extended describle (\d+), in oracle mode only
> > * Changed Regression to for (\d+)
> > * Add regression for invisible columns
> > 
> > I used this patch https://www.postgresql.org/message-id/flat/be81d947-f4fe-c062-f107-0f05f8f87ca8%40migops.com and changed it accordingly to support ivorysql in oracle mode.
> > The invisible columns support in describe command (\d+) is causing most of the test cases to change. I feel this may add difficulty during the upstream merges. If we don't need this, we can remove the invisible attribute from the describe command.
> > I didn't add the test cases with MODIFY col commands. Looks like its not working. I opened the issue here #678. I will try fixing that too.
> > Please let me know if we need to change some thing in this PR.
> > This PR is related to #608
> > ## Summary by CodeRabbit
> > 
> > * **New Features**
> >   
> >   * Introduced support for invisible columns in PostgreSQL tables.
> >   * Added operations for creating, altering, and querying tables with invisible columns.
> >   * Enhanced inheritance, partitioning, and temporary table functionalities to handle invisible columns.
> >   * Included new SQL commands and options for managing invisible columns.
> >   * Expanded capabilities for defining constraints and views with invisible columns.
> >   * Improved command completion options for altering table structures in the command line interface.
> >   * Added comprehensive test cases to showcase various scenarios involving invisible columns.
> > * **Bug Fixes**
> >   
> >   * Improved handling of column visibility during star expansion in SQL queries.
> >   * Corrected logic for attribute processing to accurately filter invisible columns.
> 
> Hello imranzaheer612, thank you for once again contributing to Ivory SQL. I spent some time reading your code and have the following two suggestions: 

Hi @jiaoshuntian , thanks for reviewing it. 

> 1.The judgment for the remoteVersion in pg_dump.c should be 170000. 

Ok I will fix that.

> 2.As you said, supporting invisible columns in “\d+” will change a lot of test cases. So, How do you think of the idea of adding a new parameter in ivy_guc to control whether it describes invisible columns. In this way, we can avoid modifying other test cases by adding this parameter's test case to the invisible.sql.

I don't think we need a separate guc variable for that, as `\d+` does a better job explaining the table attributes and its more handy. I guess its ok if we keep using describe command and test cases should be changed accordingly which I already did. Also its okay if we have to change some test cases if a feature requires it so. 

What do you say?

**Pull request merged**